File: backend/models/Task.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const TaskSchema = new Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'user'
    },
    title: {
        type: String,
        required: true
    },
    description: {
        type: String
    },
    status: {
        type: String,
        enum: ['To-Do', 'In Progress', 'Under Review', 'Finished'],
        default: 'To-Do'
    },
    priority: {
        type: String
    },
    deadline: {
        type: Date
    },
    date: {
        type: Date,
        default: Date.now
    }
});

module.exports = mongoose.model('Task', TaskSchema);

File: Task-Management-App-Flask-Backend/backend/models/task.py
from flask_mongoengine import MongoEngine
from datetime import datetime

db = MongoEngine()

class Task(db.Document):
    user = db.ReferenceField('User', required=True)
    title = db.StringField(required=True)
    description = db.StringField()
    status = db.StringField(choices=['To-Do', 'In Progress', 'Under Review', 'Finished'], default='To-Do')
    priority = db.StringField()
    deadline = db.DateTimeField()
    date = db.DateTimeField(default=datetime.utcnow)

    meta = {
        'collection': 'tasks',
        'indexes': [
            'user',
            'status',
            'priority',
            'deadline',
            'date'
        ]
    }

    def to_dict(self):
        return {
            'id': str(self.id),
            'user': str(self.user.id),
            'title': self.title,
            'description': self.description,
            'status': self.status,
            'priority': self.priority,
            'deadline': self.deadline.isoformat() if self.deadline else None,
            'date': self.date.isoformat()
        }